source/dub/packagemanager.d

```diff
@@ -297,41 +325,62 @@ class PackageManager {
 		return &iterator;
 	}
 
-
 	/** Returns a list of all package overrides for the given scope.
 	*/
-	const(PackageOverride)[] getOverrides(LocalPackageType scope_)
-	const {
-		return m_repositories[scope_].overrides;
+	deprecated const(PackageOverride)[] getOverrides(LocalPackageType scope_)
+	{
+		return getOverrides(m_defaultPaths[scope_]);
+	}
+	/// ditto
+	const(PackageOverride)[] getOverrides(Path repoPath)
+	in { assert(repoPath.absolute); }
+	body
+	{
+		repoPath.normalize();
+		return m_repos[repoPath].getOverrides();
 	}
 
 	/** Adds a new override for the given package.
 	*/
-	void addOverride(LocalPackageType scope_, string package_, Dependency version_spec, Version target)
+	deprecated void addOverride(LocalPackageType scope_, string package_, Dependency version_spec, Version target)
+	{
+		addOverride(m_defaultPaths[scope_], package_, version_spec, target);
+	}
+	/// ditto
+	deprecated void addOverride(LocalPackageType scope_, string package_, Dependency version_spec, Path target)
 	{
-		m_repositories[scope_].overrides ~= PackageOverride(package_, version_spec, target);
-		writeLocalPackageOverridesFile(scope_);
+		addOverride(m_defaultPaths[scope_], package_, version_spec, target);
 	}
 	/// ditto
-	void addOverride(LocalPackageType scope_, string package_, Dependency version_spec, Path target)
```

Comment:
The work is now done as member of Repository, but there's a Path-based function here to refer to the right member of m_repos.